File: greedy/2971_find_polygon_with_the_largest_perimeter.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var largestPerimeter = function(nums) {

    let ans = -1;
    nums = nums.sort((a,b) => a-b);

    let count = nums[0] + nums[1];

    for(let i=2; i<nums.length; i++){
        if(nums[i] < count){
            count += nums[i];
            ans = Math.max(count, ans);
        } else {
            count += nums[i];
        }
    }

    return ans;

};

File: greedy/2971_find_polygon_with_the_largest_perimeter.rs
impl Solution {
    pub fn largest_perimeter(nums: Vec<i32>) -> i64 {

        let mut ans:i64 = -1;
        let mut nums = nums;
        nums.sort();

        let mut count: i64 = (nums[0] + nums[1]) as i64;

        for i in 2.. nums.len() {

            let curr = i64::from(nums[i]);

            if curr < count {
                count += curr;
                ans = count;
            } else {
                count += curr;
            }
        }

        ans

    }
}